query/src/main/java/io/keen/client/java/MultiAnalysis.java

```diff
@@ -0,0 +1,172 @@
+        if (null == this.analyses || this.analyses.isEmpty()) {
+            throw new IllegalArgumentException("MultiAnalysis requires that 'analyses' be set.");
+        }
+    }
+
+    /**
+     * This builder class helps configure the required and optional parameters that are appropriate
+     * for a MultiAnalysis.
+     */
+    public static class Builder extends CollectionAnalysis.Builder<Builder> {
+        // required
+        private Collection<SubAnalysis> subAnalyses;
+
+
+        /**
+         * Get the collection of sub-analyses for this multi-analysis.
+         *
+         * @return The sub-analyses.
+         */
+        public Collection<SubAnalysis> getSubAnalyses() { return this.subAnalyses; }
+
+
+        // Configurable properties added by this builder.
+
+        /**
+         * Set the collection of sub-analyses. Existing sub-analyses will be discarded.
+         *
+         * @param subAnalyses The new analysis arguments to replace the discarded ones, if any.
+         */
+        public void setSubAnalyses(Collection<? extends SubAnalysis> subAnalyses) {
+            this.subAnalyses = null;
+
+            // Client code may just be clearing all the analyses.
+            if (null != subAnalyses) {
+                withSubAnalyses(subAnalyses);
+            }
+        }
+
+        /**
+         * Adds a collection of sub-analyses.
+         *
+         * @param subAnalyses The new analysis arguments to add to the existing analyses.
+         * @return This instance (for method chaining).
+         */
+        public Builder withSubAnalyses(Collection<? extends SubAnalysis> subAnalyses) {
+            if (null == subAnalyses) {
+                throw new IllegalArgumentException("The 'subAnalyses' parameter cannot be null.");
+            }
+
+            // Add each sub-analysis to the list, appending to anything that already exists.
+            for (SubAnalysis subAnalysis : subAnalyses) {
+                addSubAnalysis(subAnalysis);
+            }
+
+            return this;
+        }
+
+        /**
+         * Adds a sub-analysis to the existing analyses.
+         *
+         * @param subAnalysis The sub-analysis to add to the existing analyses.
+         * @return This instance (for method chaining).
+         */
+        public Builder withSubAnalysis(SubAnalysis subAnalysis) {
+            addSubAnalysis(subAnalysis);
+            return this;
+        }
+
+        /**
+         * Adds a group by property to the list of existing group by properties.
+         *
+         * @param subAnalysis TThe sub-analysis to add to the existing analyses.
+         */
+        public void addSubAnalysis(SubAnalysis subAnalysis) {
+            if (null == subAnalysis) {
+                throw new IllegalArgumentException("The 'subAnalysis' parameter cannot be null.");
+            }
+
+            if (null == this.subAnalyses) {
+                // No point in specifying one twice. Though obviously for now, since SubAnalysis
```

Comment:
I'm just saying that if you add the same exact analysis (and elsewhere filter also) we should throw because that's probably an error in client code, or else explicitly detect it and toss it since it'll just grow the payload of the request. HashSet, as opposed to other Collection types, imposes set semantics, so that just happens, but only if you override the hash/equality methods. I put the comment at the point where we pick the concrete type of the ```subAnalyses``` Collection since that's where the decision to use set semantics is ostensibly made. But it only really prevents us from duplicating instances with reference equality for the time being, hence the comment.